zhaquirks/onesti/easycodetouch.py

```diff
@@ -0,0 +1,99 @@
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+from zigpy.zcl.clusters.closures import DoorLock
+from zigpy.zcl.clusters.general import (
+    Basic,
+    Groups,
+    Identify,
+    Ota,
+    PowerConfiguration,
+    Scenes,
+)
+import zigpy.zdo.types
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    NODE_DESCRIPTOR,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+MANUFACTURER_SPECIFIC_PROFILE_ID = 0xFEA2
+
+
+class BatteryBasicCluster(CustomCluster, Basic):
+    """Basic cluster implementation.
+    This implementation sets power source to battery
+    and removes incorrect MainsPowered-flag.
+    """
+
+    cluster_id = Basic.cluster_id
+    POWER_SOURCE = 0x0007
```

Comment:
Getting attribute ids will be easier in the future, but right now, we can already do this:
```suggestion
    Basic.attributes_by_name["power_source"].id
```